queries/go.scm

```diff
@@ -0,0 +1,46 @@
+;; Generated by the following command:
+;; `curl https://raw.githubusercontent.com/tree-sitter/tree-sitter-go/master/src/node-types.json | jq '[.[] | select(.type == "_statement" or .type == "_simple_statement") | .subtypes[].type]'`
+[
+  (assignment_statement)
+  (dec_statement)
+  (inc_statement)
+  (send_statement)
+  (short_var_declaration)
+  (break_statement)
+  (const_declaration)
+  (continue_statement)
+  (defer_statement)
+  (empty_statement)
+  (expression_switch_statement)
+  (fallthrough_statement)
+  (for_statement)
+  (go_statement)
+  (goto_statement)
+  (if_statement)
+  (labeled_statement)
+  (return_statement)
+  (select_statement)
+  (type_declaration)
+  (type_switch_statement)
+  (var_declaration)
+] @statement
+
+(composite_literal
+  type: [
+    (map_type)
```

Comment:
On further reflection, there's more subtlety here. More details later, but we probably shouldn't merge this as-is.

Reply:
That's fine.  Do you want to take this one over, or should we pair up again?  Fine either way by me

Reply:
Probably easiest for me to take it over. I have a design question here, though.

The problem is that the type of a Go composite literal isn't a purely syntactic property. Examples:

```go
type T map[int]string
type U [2]string
type V struct{ a, b string }

var t T{1: "a"} // a map literal
var u U{1: "a"} // an array literal

var u2 U{"a", "b"}  // an array literal
var v V{"a", "b"} // a slice literal
```

There are heuristics we could use that would probably be 90-99% accurate.

* If there's an explicit syntactic type, use it. (That's what this PR does.)
* If there's an index present, call it a map. (Fails on `u` above.)
* If there's no index present, call it a list. (Fails on `v` above.)

Or we could give up and declare all composite literals to be both `map` and `list`, the current behavior.

The question is: Is it better to be (a) more precise, more complicated, and occasionally wrong, or (b) less precise, and simpler.

Reply:
how about if it has `:`s in it, it's a map, otherwise, it's a list?

Reply:
yep, that's the heuristic i alluded to above. just wanted to confirm that heuristic is better than simple. sounds like it is; i will start with that.